connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1090,6 +1090,39 @@ public void putConnectorConfig(final String connName, final Map<String, String>
         );
     }
 
+    @Override
+    public void stopConnector(final String connName, final Callback<Void> callback) {
+        log.trace("Submitting request to transition connector {} to STOPPED state", connName);
+
+        addRequest(
+                () -> {
+                    refreshConfigSnapshot(workerSyncTimeoutMs);
```

Comment:
> I do want to add a forced read-to-end after we write the new target state, though, so that the leader can correctly reject attempts to write new task configs for the connector if it is stopped.

Yeah, that makes sense 👍